src/execution_tree/compiler/compiler.cpp

```diff
@@ -466,11 +466,15 @@ namespace phylanx { namespace execution_tree { namespace compiler
                 snippets_.sequence_numbers_[name]++;
 
             // get global name of the component created
-            std::string global_name =
-                name + "$" + std::to_string(sequence_number);
+            primitive_name_parts name_parts;
+            name_parts.primitive = name;
```

Comment:
`std::move`?

Reply:
It's being used all over the place. Should we?